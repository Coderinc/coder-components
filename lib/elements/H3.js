import PropTypes from 'prop-types';
import styled from 'styled-components';

const H3 = styled.h3`
  color: ${({ theme }) => theme.primaryFont};
  font-size: 1.25rem;
  line-height: 2rem;
  font-weight: ${({ light }) => (light ? 'lighter' : 'bold')};
`;

H3.propTypes = {
  light: PropTypes.bool,
};

H3.defaultProps = {
  light: false,
};

export default H3;
