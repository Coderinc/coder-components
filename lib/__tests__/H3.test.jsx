import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import H3 from '../elements/H3';

describe('H3', () => {
  it('renders correctly', () => {
    const component = renderer.create(<H3>Lorem Ipsum</H3>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('light font', () => {
    const component = renderer.create(<H3 light>Lorem Ipsum</H3>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});