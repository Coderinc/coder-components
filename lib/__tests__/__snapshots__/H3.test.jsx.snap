// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H3 light font 1`] = `
.c0 {
  font-size: 1.25rem;
  line-height: 2rem;
  font-weight: lighter;
}

<h3
  className="c0"
>
  Lorem Ipsum
</h3>
`;

exports[`H3 renders correctly 1`] = `
.c0 {
  font-size: 1.25rem;
  line-height: 2rem;
  font-weight: bold;
}

<h3
  className="c0"
>
  Lorem Ipsum
</h3>
`;
